global/scripts/semgrep/run.sh

```diff
@@ -11,23 +11,46 @@ fileName="$CONTAINER_PATH/$REPORT_PATH/semgrep.json"
 
 defaultFile="$SCRIPTS_DIR/global/scripts/semgrep/.semgrepignore"
 cp "$defaultFile" .
-
 if [ -f ".semgrep.yaml" ]; then
-  docker run \
-  -v "$(pwd):$CONTAINER_PATH" \
-  --workdir "$CONTAINER_PATH" \
-  returntocorp/semgrep:latest \
-  semgrep \
-  --metrics=off \
-  --config "p/$1" \
-  --config "p/docker" \
-  --config "p/dockerfile" \
-  --config "p/secrets" \
-  --config "p/owasp-top-ten" \
-  --config "p/r2c-best-practices" \
-  --config ".semgrep.yaml" \
-  --enable-version-check --force-color \
-  --error --json --output "$fileName" || EXIT_CODE=$?
+  if [ -f ".semgrepcustom" ]; then
+    SEMGREP_CUSTOM=""
+    IFS=';'
+    for line in $(cat ".semgrepcustom"); do
+      SEMGREP_CUSTOM="${SEMGREP_CUSTOM}${line}"
+    done
+    docker run \
+    -v "$(pwd):$CONTAINER_PATH" \
+    --workdir "$CONTAINER_PATH" \
+    returntocorp/semgrep:latest \
+    semgrep \
+    --metrics=off \
+    --config "p/$1" \
+    --config "p/docker" \
+    --config "p/dockerfile" \
+    --config "p/secrets" \
+    --config "p/owasp-top-ten" \
+    --config "p/r2c-best-practices" \
+    --exclude-rule $SEMGREP_CUSTOM \
+    --config ".semgrep.yaml" \
+    --enable-version-check --force-color \
+    --error --json --output "$fileName" || EXIT_CODE=$?
+  else
+    docker run \
```

Comment:
Instead of adding a lot of repeated commands, you can create a string, and add the commands in the string. After you can use an "eval" to execute the command you've created concatenating the string. What do you think?

Reply:
I changed the main command to use eval and with that I was able to simplify everything. Thanks.